packages/dds/tree/src/simple-tree/core/treeNodeKernel.ts

```diff
@@ -97,39 +114,19 @@ export class TreeNodeKernel implements Listenable<KernelEvents> {
 	 */
 	public generationNumber: number = 0;
 
-	#hydrated?: {
-		anchorNode: AnchorNode;
-		offAnchorNode: Set<Off>;
-	};
+	#hydrationState: HydrationState = () => {};
 
 	/**
 	 * Events registered before hydration.
 	 * @remarks
 	 * As an optimization these are allocated lazily as they are usually unused.
```

Comment:
Nit (while we're here)
```suggestion
	 * Since these are usually not used, they are allocated lazily as an optimization.
```